random_fields/generate_field.py

```diff
@@ -91,6 +132,121 @@ def generate(self, nodes: np.ndarray) -> None:
+        mask = np.arange(len(scale_fluctuation)) != self.v_dim
+        scale_fluctuation[mask] = scale_fluctuation[mask] * self.anisotropy
+
+        # correct the length scales between libraries
+        if self.random_field_model_name == 'Gaussian':
+            ls_sklearn = scale_fluctuation * 2 / np.pi
+            self.kriging_kernel = WhiteKernel(noise_level) + RBF(length_scale=ls_sklearn)
+        elif self.random_field_model_name == 'Exponential':
+            ls_sklearn = scale_fluctuation / np.sqrt(1 / 2)
+            self.kriging_kernel = WhiteKernel(noise_level) + Matern(length_scale=ls_sklearn, nu=0.5)
+        elif self.random_field_model_name == 'Matern':
+            ls_sklearn = scale_fluctuation / np.sqrt(1 / 2)
+            self.kriging_kernel = WhiteKernel(noise_level) + Matern(length_scale=ls_sklearn)
+
+        # fit a GP against conditioning points standardised by simulation field statistics:
+        # NOT by conditioning point statistics
+        self.gaussian_process = GPR(kernel=self.kriging_kernel)
+        self.gaussian_process.optimizer = None
+        self.gaussian_process.fit(points, (values - self.mean) / np.sqrt(self.variance))
+
+    def generate_conditioned(self, nodes: npt.NDArray[np.float64]) -> None:
+        """
+        Generate conditioned random field
+
+        Args:
+            - nodes (npt.NDArray[np.float64]): The nodes of the random field
+
+        Raises:
+            - ValueError: if dimensions of nodes do not match the dimensions of the model
+        """
+
+        # check dimensions of nodes agrees with dimensions of model
+        if nodes.shape[1] != self.n_dim:
+            raise ValueError(f'Dimensions of nodes: {nodes.shape[1]} do not match dimensions of model: {self.n_dim}')
+
+        # create kriged mean field
+        z_kriged_field, std_kriged_field = self.gaussian_process.predict(nodes, return_std=True)
+
+        std_kriged_field = np.sqrt(std_kriged_field**2 - self.noise_level)
+
+        std_kriged_field *= np.sqrt(self.variance)
+        self.kriging_mean = self.mean + np.sqrt(self.variance) * z_kriged_field
+
+        # create single random field at nodes and conditioning points
+        self.generate(np.vstack([nodes, self.conditioning_points]))
+
+        # Split the generated field into the nodal coordinates and the conditioning points
+        # Standardize the distribution to marginal ~N(0,1)
+        z_cond_rf_nodes = (self.random_field.field[:nodes.shape[0]] - self.mean) / np.sqrt(  # type: ignore
+            self.variance)  # type: ignore
+        z_rf_cond_points = (self.random_field.field[nodes.shape[0]:] - self.mean) / np.sqrt(  # type: ignore
+            self.variance)  # type: ignore
+
+        # add the noise to the conditioning points
+        z_rf_cond_points += np.sqrt(self.noise_level) * np.random.normal(size=z_rf_cond_points.shape)
+
+        # create kriged mean field of random field
+        gp_rf = GPR(kernel=self.gaussian_process.kernel_)
```

Comment:
self.gaussian_process also is an instance of the class GPR(), does this second call to GPR make sense?